programs/bpf/tests/programs.rs

```diff
@@ -3524,3 +3526,120 @@ fn test_program_fees() {
     let post_balance = bank_client.get_balance(&mint_keypair.pubkey()).unwrap();
     assert_eq!(pre_balance - post_balance, expected_min_fee);
 }
+
+#[test]
+#[cfg(feature = "bpf_rust")]
+fn test_get_minimum_delegation() {
+    solana_logger::setup();
+
+    let GenesisConfigInfo {
+        genesis_config,
+        mint_keypair,
+        ..
+    } = create_genesis_config(100_123_456_789);
+    let mut bank = Bank::new_for_tests(&genesis_config);
+    bank.feature_set = Arc::new(FeatureSet::all_enabled());
+
+    let (name, id, entrypoint) = solana_bpf_loader_program!();
+    bank.add_builtin(&name, &id, entrypoint);
+    let bank = Arc::new(bank);
+    let bank_client = BankClient::new_shared(&bank);
+
+    let program_id = load_bpf_program(
+        &bank_client,
+        &bpf_loader::id(),
+        &mint_keypair,
+        "solana_bpf_rust_get_minimum_delegation_data",
+    );
+
+    let stake_keypair = Keypair::new();
+    let stake_account = AccountSharedData::new(123_456_789, 0, &stake::program::id());
+    bank.store_account(&stake_keypair.pubkey(), &stake_account);
+
+    // bprumo TODO: what accounts are actually needed here? stake::program and stake_keypair both
+    // seem to be needed, but the test still errors saying there's insufficient account keys for
+    // the instruction
+    let account_metas = vec![
+        //AccountMeta::new(mint_keypair.pubkey(), true),
+        // bprumo NOTE: without this account: NotEnoughAccountKeys `failed: insufficient account keys for instruction`
+        AccountMeta::new(stake_keypair.pubkey(), false),
+        // bprumo NOTE: without this account: MissingAccount `failed: An account required by the instruction is missing`
+        AccountMeta::new_readonly(stake::program::id(), false),
+    ];
+
+    // bprumo TODO: is `data` needed here? &[] and &[1] have the same result/error
```

Comment:
Typically programs use the first byte or first four bytes to determine which instruction is being invoked (this is where the stake instruction enum comes into play). So this data needs to encode the bytes for your new ix

Reply:
Doesn't the data here refer to what's passed to my example program? Inside my example program I'm creating the GetMinimumDelegation instruction and passing that directly to invoke. Maybe this is the root of my issue?

https://github.com/solana-labs/solana/blob/1ba3960af1e215d266c27576184d8abe5814d654/programs/bpf/rust/get_minimum_delegation_data/src/lib.rs#L32-L33

Reply:
Ah sorry my mistake, got confused, you're right. You don't need to pass any data to your program